statsmodels/tsa/statespace/mlemodel.py

```diff
@@ -1004,9 +1004,10 @@ def _score_obs_harvey(self, params, approx_complex_step=True,
         partials = np.zeros((self.nobs, n))
         k_endog = self.k_endog
         for t in range(self.nobs):
-            for i in range(n):
-                inv_forecasts_error_cov = np.linalg.inv(
+            inv_forecasts_error_cov = np.linalg.inv(
                     res.forecasts_error_cov[:, :, t])
+
```

Comment:
Noticed this while skimming: `inv_forecasts_error_cov` is re-calculated in each step of the loop over `i`.  The diff is a bit hard to read; it just moves that calculation outside the loop.  There are some other repeated calculations inside this loop, but they don't get stored as intermediate variables so it is _conceivable that there is a memory tradeoff.  Leaving this with a big "moderately-low hanging fruit" sign attached.

Reply:
Yes, I think this looks good.